connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -128,6 +151,26 @@ private[sql] class AvroDeserializer(
       case (INT, IntegerType) => (updater, ordinal, value) =>
         updater.setInt(ordinal, value.asInstanceOf[Int])
 
+      case (INT, DateType) if isNotExactType =>
+        throw incorrectTypeException(DateType)
+
+      case (LONG, DateType) if isNotExactType =>
+        throw new IncompatibleSchemaException(incompatibleMsg)
+
+      case (INT, y: YearMonthIntervalType) if isNotExactType =>
```

Comment:
Also, why do we have to block this? 
As per https://github.com/apache/spark/commit/562f43c945315aaa0d0979ec1a16bfb5616dd053, Avro should be able to read/write ansi interval types.
cc @MaxGekk

Reply:
Yeah, I don't need to block this